backend/static/js/codemirror/modes/sparql/sparql-hint.js

```diff
@@ -717,18 +717,19 @@ function parseAndEvaluateCondition(condition, word, lines, words) {
 }
 
 // Get result of SPARQL query with timeout.
-//
-// TODO: Note that the QLever timeout can have a delay. In particular, this is
-// relevant for mixed AC queries. For example, when we specify a timeout of 1.0s
-// for the sensitive AC query, and QLever takes 1.7s before timing out, then the user
-// has to wait 1.7s for their suggestions.
-const fetchTimeout = async (sparqlQuery, timeoutSeconds, queryId) => {
+const fetchTimeout = (sparqlQuery, timeoutSeconds, queryId) => {
   const parameters = {
     query: sparqlQuery,
-    // Only add timeout if it's positive
-    ...(timeoutSeconds > 0 && { timeout: `${timeoutSeconds}s` })
+    // Only add timeout if it's positive. QLever only supports
+    // second-level precision for timeout values, so round up to
+    // the nearest integer.
+    ...(timeoutSeconds > 0 && { timeout: `${Math.ceil(timeoutSeconds)}s` })
   };
-  return fetchQleverBackend(parameters, { "Query-Id": queryId });
+  return Promise.race([
+    // Don't wait for QLever after the timeout expired.
+    new Promise((resolve) => setTimeout(resolve, timeoutSeconds * 1000, { exception: 'Timeout reached' })),
+    fetchQleverBackend(parameters, { "Query-Id": queryId })
+  ]);
```

Comment:
Is this for the case that QLever takes much longer than the specified timeout? Whatever it is, explain more clearly in the comment, why this is needed